lib/components/dashboard/HandlePlanShareDummy.tsx

```diff
@@ -124,30 +129,19 @@ const HandlePlanShareDummy = () => {
 
   /**
    * Caches all courses in plans
-   * @param years - an array of years of the plan
+   * @param id - plan id
    */
-  const cache = (years: Year[]) => {
-    let total = 0;
-    let cum = 0;
-    years.forEach((y) => {
-      if (cum === total) {
-        setCached(true);
-      }
-      y.courses.forEach((c) => {
-        total++;
-        axios
-          .get(getAPI(window) + '/search', {
-            params: { query: c._id },
-          })
-          .then((retrieved) => {
-            dispatch(updateCourseCache(retrieved.data.data));
-            cum++;
-            if (cum === total) {
-              setCached(true);
-            }
-          });
+  const cache = (id: string) => {
+    axios
+      .get(getAPI(window) + '/coursesByPlan/' + id)
+      .then((response) => {
+        const sisCourses: SISRetrievedCourse[] = response.data.data;
+        dispatch(updateCourseCache(sisCourses));
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
This functions takes Course obj in frontend to query for a SIS Course with version. 

before: take in an array of years and cache courses one by one with `/search` route 
after: make one API call to get all SISCourseV documents associated with specified plan